### PR TITLE
Python: add new Python features

### PR DESCRIPTION
**Pull request by maximiliankaul**
This branch adds new Python features and fixes bugs identified with the code snippet in #432 
This is not a complete fix, as we currently do not store expressions / statements outside of records / functions in the cpg (see issue #464)
Changes:
- return statements can now hold a value
- lists of statements are now either wrapped in a compound statement (size > 1) or directly returned
- more tests

### CODE REVIEW

**Comment by oxisto** [comment]
Is there a particular reason to not wrap statement of length in a CompoundStatement. Is it just for convinence or did it cause any of the bugs in the mentioned issue? Not 100% sure, how consistently we do this in other languages. The CPG structure explicitly allows this, so I guess it should be ok.

File: cpg-library/src/main/python/cpg.py [hunk 1]
```diff
@@ -1178,6 +1194,7 @@ def visit_ClassDef(self, node):
                 # TODO what to do about expressions inside a class?
                 self.visit(b)
             else:
+                print(b)
```
Comment (oxisto) [inline cpg-library/src/main/python/cpg.py:1197]: Remove print

File: cpg-library/src/test/java/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 2]
```diff
@@ -506,4 +499,104 @@ class PythonFrontendTest : BaseTest() {
...
+        assertEquals(clsC1.fields.size, 1)
+        val field = clsC1.fields[0]
+        assertNotNull(field)
+        assertEquals(field.name, "total")
+        // TODO assert initializer "total = 0"
+        val meth = clsC1.methods[0]
+        assertNotNull(meth)
+        assertEquals(meth.name, "inc")
+        assertEquals(meth.recordDeclaration, clsC1)
+        val selfReceiver = meth.receiver
```
Comment (oxisto) [inline cpg-library/src/test/java/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:573]: some newlines now and then would be good for readability.

**Comment by maximiliankaul** [comment]
> Is there a particular reason to not wrap statement of length == 1 in a CompoundStatement? Is it just for convinence or did it cause any of the bugs in the mentioned issue? Not 100% sure, how consistently we do this in other languages. The CPG structure explicitly allows this, so I guess it should be ok.

I just thought this would be nicer to work with. There is no reason that I can think of not to wrap single statements in a CompoundStatement. Do you prefer to have everything in a CompoundStatement?

**Comment by oxisto** [comment]
> > Is there a particular reason to not wrap statement of length == 1 in a CompoundStatement? Is it just for convinence or did it cause any of the bugs in the mentioned issue? Not 100% sure, how consistently we do this in other languages. The CPG structure explicitly allows this, so I guess it should be ok.
> 
> I just thought this would be nicer to work with. There is no reason that I can think of not to wrap single statements in a CompoundStatement. Do you prefer to have everything in a CompoundStatement?

I do not have a strong preference for or against it I would say. The only problem we could run into is if one manages to declare variables within the one statement and also uses it. Not sure if this is possible. Because variable resolution depends on a scope (of the compound statement) and with just a single statement, there is also no scope.

**Comment by maximiliankaul** [comment]
> > > Is there a particular reason to not wrap statement of length == 1 in a CompoundStatement? Is it just for convinence or did it cause any of the bugs in the mentioned issue? Not 100% sure, how consistently we do this in other languages. The CPG structure explicitly allows this, so I guess it should be ok.
> > 
> > 
> > I just thought this would be nicer to work with. There is no reason that I can think of not to wrap single statements in a CompoundStatement. Do you prefer to have everything in a CompoundStatement?
> 
> I do not have a strong preference for or against it I would say. The only problem we could run into is if one manages to declare variables within the one statement and also use it. Not sure if this is possible. Because variable resolution depends on a scope (of the compound statement) and with just a single statement, there is also no scope.

I would prefer to have it as it is currently implemented. And it will be an easy fix, if we decide to change this in the future as only the function `make_compound_statement` (currently a very misleading name...) needs to be adjusted.

**Review by oxisto — APPROVED** [review]

**Comment by maximiliankaul** [comment]
I would like to merge this PR as some of my other work depends on this. However, there is still the bug with statements outside of functions / records that depends on #464 .

**Comment by oxisto** [comment]
> I would like to merge this PR as some of my other work depends on this. However, there is still the bug with statements outside of functions / records that depends on #464 .

Agreed. We will merge this, so you can continue

**Pull request merged**